packages/synapse-react-client/src/components/WikiMarkdownEditor/WikiMarkdownEditor.tsx

```diff
@@ -0,0 +1,256 @@
+import { Alert, Box, Button, TextField } from '@mui/material'
+import { ObjectType, WikiPageKey } from '@sage-bionetworks/synapse-types'
+import { noop } from 'lodash-es'
+import React, { useEffect, useMemo, useState } from 'react'
+import {
+  CreateWikiPageInput,
+  UpdateWikiPageInput,
+  useCreateWikiPage,
+  useGetRootWikiPageKey,
+  useGetWikiPage,
+  useUpdateWikiPage,
+} from '../../synapse-queries'
+import ConfirmationDialog from '../ConfirmationDialog'
+import { DialogBase } from '../DialogBase'
+import { SynapseSpinner } from '../LoadingScreen/LoadingScreen'
+import { MarkdownEditor } from '../Markdown'
+
+export const UNSAVED_CHANGES = 'Unsaved Changes'
+export const NAVIGATE_AWAY_CONFIRMATION_MESSAGE =
+  'Any unsaved changes may be lost. Are you sure that you would like to navigate away from this editor?'
+export const ERROR_LOADING_WIKI_FAILED = 'Failed to load the wiki page: '
+export const ERROR_SAVING_WIKI =
+  'Could not save your changes. It is recommended that you copy your version of the wiki text so that it is not lost. '
+
+export type WikiMarkdownEditorProps = {
+  ownerObjectType: ObjectType
+  ownerObjectId: string
+  // if wikiPageId is undefined, will get (or create) the root WikiPage for this ownerObject
+  // otherwise, will get the WikiPage with the specified wikiPageId
+  wikiPageId?: string
+  onCancel?: () => void
+  onSave?: () => void
+  // TODO: SWC-6774 - expose delete prop
+  // showDeleteButton: boolean
+}
+
+export const WikiMarkdownEditor: React.FunctionComponent<
+  WikiMarkdownEditorProps
+> = (props: WikiMarkdownEditorProps) => {
+  // TODO: SWC-6774 - get showDeleteButton from props
+  const showDeleteButton = false
+  const {
+    ownerObjectType,
+    ownerObjectId,
+    wikiPageId: initialWikiPageId,
+    onCancel = noop,
+    onSave = noop,
+  } = props
+
+  const [wikiPageId, setWikiPageId] = useState<string | undefined>(
+    initialWikiPageId,
+  )
+  const [open, setOpen] = useState<boolean>(true)
+  const [title, setTitle] = useState<string>('')
+  const [markdown, setMarkdown] = useState<string>('')
+  const [showConfirmCancelDialog, setShowConfirmDialog] =
+    useState<boolean>(false)
+  const [error, setError] = useState<string | null>(null)
+
+  const { data: rootWikiPageKey, error: errorLoadingRootWikiPageKey } =
+    useGetRootWikiPageKey(ownerObjectType, ownerObjectId, {
+      enabled: wikiPageId === undefined,
+    })
```

Comment:
When a specific `wikiPageId` is not provided, get the `WikiPageKey` for the root `WikiPage`.

Reply:
Suspect that the SWC implementation follows this path because there are services to GET a specific WikiPage by id for [ENTITY](https://rest-docs.synapse.org/rest/GET/entity/ownerId/wiki/wikiId.html), [EVALUATION](https://rest-docs.synapse.org/rest/GET/evaluation/ownerId/wiki/wikiId.html), and [ACCESS_REQUIREMENT](https://rest-docs.synapse.org/rest/GET/access_requirement/ownerId/wiki/wikiId.html) owner object types. However, there are only services to get the root WikiPage for [ENTITY](https://rest-docs.synapse.org/rest/GET/entity/ownerId/wiki.html) and [EVALUATION](https://rest-docs.synapse.org/rest/GET/evaluation/ownerId/wiki.html), not for ACCESS_REQUIREMENT. 

So, when a specific wikiPageId is provided, we can directly GET the WikiPage. However, when a wikiPageId is not provided, then first get the root WikiPageKey (since there are services to get root WikiPageKey for [ENTITY](https://rest-docs.synapse.org/rest/GET/evaluation/ownerId/wikikey.html), [EVALUATION](https://rest-docs.synapse.org/rest/GET/entity/ownerId/wiki/wikiId.html), AND [ACCESS_REQUIREMENT](https://rest-docs.synapse.org/rest/GET/access_requirement/ownerId/wikikey.html) owner object types), then get the root WikiPage by id.